File: Go/day1_part1.go
package main

import (
	"fmt"
	"io/ioutil"
)

func day1Part1() {
	in, _ := ioutil.ReadFile("../day1_input.txt")
	s := string(in)
	total := 0
	for i, curr := range s {
		if curr == rune(s[(i+1)%len(s)]) {
			total += int(curr - '0')
		}
	}
	fmt.Println(total)
}

File: Kotlin/src/day1_part1.kt
import java.io.File

fun main(args: Array<String>) {
    var total = 0
    val inputStream = File("../day1_input.txt").inputStream()
    val s = inputStream.bufferedReader().use { it.readText() }

    s.forEachIndexed { i, j ->
        if (j == s.get((i+1)%s.length)){
            total += j.toInt()-48
        }
    }
    println(total)
}